tests/phpunit/api/v3/MembershipTypeTest.php

```diff
@@ -141,8 +141,9 @@ public function testUpdateWithoutId() {
       'visibility' => 'public',
     );
 
-    $membershipType = $this->callAPIFailure('membership_type', 'create', $params);
-    $this->assertEquals($membershipType['error_message'], 'Mandatory key(s) missing from params array: domain_id');
+    $membershipType = $this->callAPISuccess('membership_type', 'create', $params);
+    $domainID = $this->callAPISuccessGetValue('MembershipType', ['return' =>'domain_id', 'id' => $membershipType['id']]);
+    $this->assertEquals(CRM_Core_Config::domainID(), $domainID);
```

Comment:
@eileenmcnaughton I feel like we need a test where x membership type has been created with domain y but now we are on domain z and do a create call and make sure that the type stays as domain y

Reply:
yeah good idea - to stop anyone else breaking later

Reply:
I've encountered an issue where a membership renewal made via iATS's scheduled job on domain Y renewing a membership on domain Z caused some havoc - I imagine IPNs would be the same.  It seemed non-trivial to fix though, and it's more a misconfiguration than a bug.

Reply:
ug - that hurts my head - only tangentally related to this though?